homeassistant/components/refoss/__init__.py

```diff
@@ -0,0 +1,95 @@
+"""Refoss devices platform loader"""
+from __future__ import annotations
+
+from typing import Collection
+from datetime import timedelta
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MAC
+from homeassistant.core import HomeAssistant
+from refoss_ha.http_device import HttpDeviceInfo
+from refoss_ha.const import (
+    DOMAIN,
+    LOGGER,
+    DEVICE_LIST_COORDINATOR,
+    SOCKET_DISCOVER_UPDATE_INTERVAL,
+    PLATFORMS,
+)
+from .coordinator import RefossCoordinator
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    if not config_entry.data.get(CONF_MAC):
+        LOGGER.warning(
+            (
+                "The config entry %s probably comes from a custom integration, please"
+                " remove it if you want to use core refoss integration"
+            ),
+            config_entry.title,
+        )
+        return False
+    hass.data[DOMAIN] = {}
+    refoss_coordinator = RefossCoordinator(
+        hass=hass,
+        config_entry=config_entry,
+        update_interval=timedelta(seconds=SOCKET_DISCOVER_UPDATE_INTERVAL),
+    )
+    try:
+        await refoss_coordinator.initial_setup()
+    except Exception as e:
+        LOGGER.warning("initial_setup failed: %s", e)
+        return False
+
+    hass.data[DOMAIN]["ADDED_ENTITIES_IDS"] = set()
+
+    hass.data[DOMAIN][DEVICE_LIST_COORDINATOR] = refoss_coordinator
```

Comment:
Ideally domain data is a dataclass

Example in `lookin/models.py`

Reply:
What's the problem with adding variables here？

Reply:
Not found  'lookin/models.py'  in the project.